src/errors.ts

```diff
@@ -52,58 +52,26 @@ export class PrivateKeyNoBackupError extends SdkError {
 }
 
 export class MultipleCardsError extends SdkError {
-    constructor(identity: string) {
+    constructor(public identity: string) {
         super(
             `There are several public keys registered with ${identity}, which is not supported.`,
             'MultipleCardsError',
         );
     }
 }
-type LookupRejected = { identity: string; error: Error };
-type LookupResolved = { identity: string; publicKey: VirgilPublicKey };
-
-export class LookupError extends SdkError {
-    result: Array<VirgilPublicKey | Error>;
-    identities: Array<string>;
 
-    rejected(): LookupRejected[] {
-        const result: LookupRejected[] = [];
-        for (let i = 0; i < this.identities.length; i++) {
-            const value = this.result[i];
-            if (value instanceof Error) {
-                result.push({
-                    identity: this.identities[i],
-                    error: value,
-                });
-            }
-        }
-        return result;
-    }
+type LookupResultWithErrors = import('./EThree').LookupResultWithErrors;
 
-    resolved(): LookupResolved[] {
-        const result: LookupResolved[] = [];
-        for (let i = 0; i < this.identities.length; i++) {
-            const value = this.result[i];
-            if (value instanceof VirgilPublicKey) {
-                result.push({
-                    identity: this.identities[i],
-                    publicKey: value,
-                });
-            }
-        }
-        return result;
-    }
-
-    constructor(identities: string[], result: Array<VirgilPublicKey | Error>) {
+export class LookupError extends SdkError {
+    constructor(public lookupResult: LookupResultWithErrors) {
         super(
-            `Failed some public keys lookups. You can see the results by calling error.resolved() and error.rejected() methods of this error instance`,
+            `Failed some public keys lookups. You can see the results by calling error.lookupResul property of this error instance`,
             'LookupError',
         );
-        this.result = result;
-        this.identities = identities;
         console.error(
```

Comment:
typically logging errors to console is only done in "dev" or "debug" mode. Keep it as is for now, but we will need to take care of this in the future versions.